src/hooks/useOperatorOrder.ts

```diff
@@ -1,30 +1,74 @@
 import { useCallback, useEffect, useState } from 'react'
 
-import { Order, getOrder } from 'api/operator'
+import { Order, getOrder, RawOrder } from 'api/operator'
 
 import { transformOrder } from 'utils'
 
 import { useNetworkId } from 'state/network'
 
 import { useMultipleErc20 } from './useErc20'
+import { Network } from 'types'
+
+const NETWORK_ID_SEARCH_LIST = [Network.Mainnet, Network.xDAI, Network.Rinkeby]
 
 type UseOrderResult = {
   order: Order | null
   error?: string
   isLoading: boolean
+  errorOrderPresentInNetworkId: Network | null
   forceUpdate: () => void
 }
 
-export function useOrder(orderId: string, updateInterval = 0): UseOrderResult {
+interface GetOrderResult {
+  order: RawOrder | null
+  errorOrderPresentInNetworkId?: Network
+}
+
+async function _getOrder(
+  networkId: Network,
+  orderId: string,
+  networkIdSearchListRemaining: Network[] = NETWORK_ID_SEARCH_LIST,
+): Promise<GetOrderResult> {
+  // Get order
+  const order = await getOrder({ networkId, orderId })
+
+  if (order || networkIdSearchListRemaining.length === 0) {
+    // We found the order in the right network
+    // ...or we have no more networks in which to continue looking
+    // so we return the "order" (can be null if it wasn't found in any network)
+    return { order }
+  }
+
+  // If we didn't find the order in the current network, we look in different networks
+  const [nextNetworkId, ...remainingNetworkIds] = networkIdSearchListRemaining.filter((network) => network != networkId)
+
+  // Try to get the oder in another network (to see if the ID is OK, but the network not)
+  const isOrderInDifferentNetwork = await getOrder({ networkId: nextNetworkId, orderId }).then(
+    (order) => order !== null,
```

Comment:
are `{}` orders accepted? is falsy all bad? or just `null`?
could be written as `.then(Boolean)`